Deploy OP Stack onto sequencer docker compose

nomaxg opened the pull request:
- Introduces a new docker compose configuration that connects an instance of the OP espresso demo stack to an existing sequencer network.
- Added a check to ensure that the corrected L1 origin is in the correct timeframe. Without this check, there was a bug where the L2 would occasionally correct to an L1 origin too far in the future. 
- Currently only works with a geth instance: [espresso sequencer geth](https://github.com/EspressoSystems/espresso-sequencer/compare/geth-image?expand=1)

Inphi committed:
feat(ctb): Rework FaultDisputeGame resolution

Initial changes to the FDG resolution function to be incentive compatible
with bonds.

Commit:
:bug: fix(indexer): add retries to db connection

maurelian committed:
feat(ctb): DelayedVetoable as a simple forwarder

Adds a very minimal DelayedVetoable, along with tests for the _handleCall function.

feat(ctb): Add actor to invariant tests

feat(ctb): Simplify by reducing the amount of assembly used.

feat(ctb): remove invariant tests

These tests have proven complex to write due to the transparent proxy pattern.

I believe that the fuzz testing is sufficient for now.

maurelian committed:
feat(op-bindings): Add bindings for DelayedVetoable

rebuild mips_more and preimageoracle_more bindings

maurelian committed:
feat(ctb): Add the delay.

maurelian committed:
feat(ctb): Add vetoer and initiator roles

feat(ctb): Add getters to DelayedVetoable

maurelian committed:
feat(ctb): move DelayedVetoable to L1 dir

maurelian committed:
feat(ctb): Add semver

ctb: Address feedback

maurelian committed:
refactor(ctb): Remove the veto function.

This simplifies the interface, and the resulting handleCall
function is also cleaner, as it treats the dispatching primarily
based on the caller.

maurelian committed:
feat(ctb): make vetoer and iniator getters virtual

This will facilitate reusability of this contract, as a future change will put the

these values in to another contract.

maurelian committed:
chore(ctb): Update semver and import path

maurelian committed:
feat(ctb): Make storage variables immutable

This contract is not intended to be upgradeable.

maurelian committed:
feat(ctb): Remove ability to receive eth

We don't expect to need this, and it adds complexity to the message digest which
we'd prefer to avoid.

maurelian committed:
feat(ctb): Start with a delay of zero and forward instantly

This allows us to include this contract in the initial deployment without having to

swap it in later.

feat(ctb): Add missing annotations

Commit:
chore(indexer): remove indexer all command

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-unicorn from 42.0.0 to 48.0.1

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 42.0.0 to 48.0.1.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v42.0.0...v48.0.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-unicorn
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

maurelian committed:
chore(ctb): Allow vetoing after the delay

test(ctb): Some DV test clean up

maurelian committed:
fix(ctb): Address compiler warnings

fix(ctb): Fix names on fuzz tests

Commit:
fix(indexer): Fix uri util

maurelian committed:
fix(ctb): Immutables scream like a snake

felipe-op committed:
feat(proxyd): high availability

felipe-op committed:
revert weird comment replaces

felipe-op committed:
semgrep

felipe-op committed:
release lock on errors, validate if local is behind remote before acquiring the lock

maurelian committed:
fix(ctb): Properly handle return data

Also adds a test that checks the return data.

maurelian committed:
forge install: safe-contracts

v1.4.1

forge install: safe-contracts

v1.4.1

maurelian committed:
refactor(ctb): Inline the _activateDelay logic

This change will enable forwarding zero length data

maurelian committed:
feat(ctb): Update semver to 1.0.0

refcell committed:
don't schedule updates for completed games

Commit:
bump version

Commit:
fix: Fix json tags

refcell committed:
add concurrency metrics

refcell committed:
labeling issue

refcell committed:
Track the number of claims in each game.

refcell committed:
zero out the claim count upon resolution

Commit:
Revert "Merge pull request #6863 from ethereum-optimism/08-17-chore_indexer_Disable_flaky_indexer_tests"

This reverts commit 306979114416fec040162506d1c927a3f3b2224b, reversing
changes made to 6f36f623eb0924244d63da346fc0770b057a08ef.

felipe-op committed:
nits

maurelian committed:
locks and bindings

felipe-op committed:
nits

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.5.0 to 20.6.1

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.5.0 to 20.6.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
feat(indexer): Enable devnet indexer as a preset

refcell committed:
introduce the rollup rpc url flag

refcell committed:
trace provider

refcell committed:
Add support for the output cannon trace type

refcell committed:
Rollup RPC Flag

Commit:
comments

Commit:
fix: Include preset in presets

Inphi committed:
op-program: Use hardcoded chain config when available

Fixes a bug where the host incorrectly informs the client that a
custom chain config is in use. The client then attempts to read the
chain config pre-image data. However, this pre-image data will not be
available in on-chain contexts.

cynic-1 committed:
Fix a typo in derivation.md

ven -> even

cynic-1 committed:
Fix a typo in glossary.md

it's -> its

karlb committed:
Support old Make version by avoiding SHELLSTATUS

SHELLSTATUS is only available on GNU Make 4.2 (released on 2016-05-22)
and later, which is unfortunatly not installed on MacOS by default.

While I am highly annoyed that Apple is unwilling to update their Make
version and don't want encourage such behaviour, the change in this
commit avoids this requirement and seems even simpler than the original
code.

refcell committed:
Address coordinator changes

mergify[bot] committed:
Merge branch 'develop' into 09-14-fix_indexer_Fix_uri_util

refcell committed:
Remove mutex and reorder metric updates

dependabot[bot] committed:
build(deps): bump ethereum/client-go in /ops-bedrock (#7283)

Bumps ethereum/client-go from v1.13.0 to v1.13.1.

---
updated-dependencies:
- dependency-name: ethereum/client-go
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mergify[bot] committed:
Merge branch 'develop' into 09-14-fix_indexer_Fix_uri_util

OptimismBot committed:
Merge pull request #7251 from ethereum-optimism/09-14-fix_indexer_Fix_uri_util

fix(indexer): Fix uri util

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/types/node-20.6.1

OptimismBot committed:
Merge pull request #7268 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.6.1

build(deps-dev): bump @types/node from 20.5.0 to 20.6.1

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/eslint-plugin-unicorn-48.0.1

trianglesphere committed:
Merge pull request #7284 from cynic-1/patch-3

Fix a typo in glossary.md

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/eslint-plugin-unicorn-48.0.1

maurelian committed:
feat(ctb): Update tests and add comments for queuedAt selector clash

maurelian committed:
bindings: Add Safe and SafeProxyFactory bindings

maurelian committed:
ci: Update semgrepignore

The trailing '/' is consistent with other directories

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.5.0 to 20.6.2

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.5.0 to 20.6.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

trianglesphere committed:
Merge pull request #7280 from cynic-1/patch-2

Fix a typo in derivation.md

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/eslint-plugin-unicorn-48.0.1

OptimismBot committed:
Merge pull request #7006 from ethereum-optimism/dependabot/npm_and_yarn/eslint-plugin-unicorn-48.0.1

build(deps-dev): bump eslint-plugin-unicorn from 42.0.0 to 48.0.1

mergify[bot] committed:
Merge branch 'develop' into refcell/output-trace-type

mergify[bot] committed:
Merge branch 'develop' into felipe/proxyd-ha

OptimismBot committed:
Merge pull request #7252 from ethereum-optimism/felipe/proxyd-ha

feat(proxyd): high availability

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/types/node-20.6.2

dependabot[bot] committed:
build(deps-dev): bump markdownlint from 0.31.0 to 0.31.1

Bumps [markdownlint](https://github.com/DavidAnson/markdownlint) from 0.31.0 to 0.31.1.
- [Changelog](https://github.com/DavidAnson/markdownlint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/DavidAnson/markdownlint/compare/v0.31.0...v0.31.1)

---
updated-dependencies:
- dependency-name: markdownlint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7295 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.6.2

build(deps-dev): bump @types/node from 20.5.0 to 20.6.2

mergify[bot] committed:
Merge branch 'develop' into refcell/output-trace-type

dependabot[bot] committed:
build(deps-dev): bump vitest from 0.34.1 to 0.34.4

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 0.34.1 to 0.34.4.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v0.34.4/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.4.0 to 6.7.2

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.4.0 to 6.7.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.2/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7273 from ethereum-optimism/refcell/output-trace-type

feat(op-challenger): Output Cannon Trace Type

OptimismBot committed:
Merge pull request #7293 from ethereum-optimism/jm/add-safe-bindings

Add bindings for Safe.sol

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/vitest-0.34.4

refcell committed:
Fix nits

Inphi committed:
op-node: Fix race condition while closing OpNode (#7303)

Before the `OpNode` closes it closes its data sources. If the runtime
config reloader goroutine is still using the data sources then it
may creates a goroutine panic. Specifically, when a `limitClient` is
used, it will be fail to write an in-flight request to its semaphore
after the client is closed.

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/vitest-0.34.4

OptimismBot committed:
Merge pull request #7297 from ethereum-optimism/dependabot/npm_and_yarn/vitest-0.34.4

build(deps-dev): bump vitest from 0.34.1 to 0.34.4

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/markdownlint-0.31.1

OptimismBot committed:
Merge pull request #7294 from ethereum-optimism/dependabot/npm_and_yarn/markdownlint-0.31.1

build(deps-dev): bump markdownlint from 0.31.0 to 0.31.1

mergify[bot] committed:
Merge branch 'develop' into jm/basic-delay

OptimismBot committed:
Merge pull request #7121 from ethereum-optimism/jm/basic-delay

Add the DelayedVetoable contract

mergify[bot] committed:
Merge branch 'develop' into inphi/host-chainid

Commit:
cleaner

Commit:
cleaner again

OptimismBot committed:
Merge pull request #7278 from ethereum-optimism/inphi/host-chainid

op-program: Use hardcoded chain config when available

mergify[bot] committed:
Merge branch 'develop' into refcell/l2-consensus

OptimismBot committed:
Merge pull request #7274 from ethereum-optimism/refcell/l2-consensus

feat(op-challenger): Rollup RPC Challenger Flag

roninjin10 committed:
Merge pull request #7091 from ethereum-optimism/09-01-fix_indexer_add_retries_to_db_connection

:bug: fix(indexer): add retries to db connection

Inphi committed:
op-challenger: Updated for the revised resolution (#7155)

* op-challenger: Update to the reworked fdg resolution

* uint64 resolve claim index

* Update op-e2e/e2eutils/disputegame/game_helper.go

Co-authored-by: refcell.eth <abigger87@gmail.com>

* Update op-e2e/e2eutils/disputegame/game_helper.go

Co-authored-by: refcell.eth <abigger87@gmail.com>

* Retry claims resolution until done

* short-circuit tryResolveClaims for empty games

* bad switch case ordering

---------

Co-authored-by: refcell.eth <abigger87@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.5.0 to 20.6.2 (#7302)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.5.0 to 20.6.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

mergify[bot] committed:
Merge branch 'develop' into inphi/fdg-rez

Commit:
fix(op-bootnode): add missed flags and enable metrics server (#7282)

mergify[bot] committed:
Merge branch 'develop' into inphi/fdg-rez

refcell committed:
Chore: Cleanup the op-bootnode.

nomaxg committed:
espresso demo setup

OptimismBot committed:
Merge pull request #7061 from ethereum-optimism/inphi/fdg-rez

feat(ctb): Rework FaultDisputeGame resolution

maurelian committed:
fix(ctb): Add vm.assume to prevent clashes on additional test cases (#7308)

* fix(ctb): Add vm.assume to prevent clashes on additional test cases

* Update DelayedVetoable.t.sol

Co-authored-by: refcell.eth <abigger87@gmail.com>

* Gas snapshot update

---------

Co-authored-by: refcell.eth <abigger87@gmail.com>
Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

OptimismBot committed:
Merge pull request #7296 from ethereum-optimism/dependabot/npm_and_yarn/typescript-eslint/parser-6.7.2

build(deps-dev): bump @typescript-eslint/parser from 6.4.0 to 6.7.2

dependabot[bot] committed:
build(deps): bump viem from 1.3.1 to 1.11.0

Bumps [viem](https://github.com/wagmi-dev/viem) from 1.3.1 to 1.11.0.
- [Release notes](https://github.com/wagmi-dev/viem/releases)
- [Changelog](https://github.com/wagmi-dev/viem/blob/main/CHANGELOG.md)
- [Commits](https://github.com/wagmi-dev/viem/compare/viem@1.3.1...viem@1.11.0)

---
updated-dependencies:
- dependency-name: viem
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7309 from ethereum-optimism/dependabot/npm_and_yarn/viem-1.11.0

build(deps): bump viem from 1.3.1 to 1.11.0

mergify[bot] committed:
Merge branch 'develop' into refcell/cleanup-bootnode

OptimismBot committed:
Merge pull request #7307 from ethereum-optimism/refcell/cleanup-bootnode

chore: Cleanup the bootnode Makefile and Entrypoint

OptimismBot committed:
Merge pull request #7286 from celo-org/karlb/makefile-macos-fix

Support old Make version by avoiding SHELLSTATUS

refcell committed:
Constant time parent lookup fix.

nomaxg committed:
fix l1 origin bug

nomaxg committed:
remove unecessary log

nomaxg commented on bedrock-devnet/devnet/__init__.py:
```diff
@@ -15,25 +15,77 @@
+)
+parser.add_argument(
+    "--allocs",
+    help="Only create the allocs and exit",
```
@jbearer our auto-formatters appear to be configured differently, let me know your settings and I'll adjust

jbearer replied:
I don't think I ran a formatter, so we can go with yours. How did you set it up?

nomaxg replied:
I use VSCode, it's just using the default [Black Formatter](https://github.com/psf/black)

jbearer replied:
Actually, many of these changed lines are from the upstream optimism repo. So in the interest of keeping the changes in our fork to a minimum maybe it's better if we don't format this file?

nomaxg commented on bedrock-devnet/devnet/__init__.py:
```diff
@@ -102,69 +158,112 @@ def main():
+        ["cast", "balance", "0x3fAB184622Dc19b6109349B94811493BF2a45362"],
+        capture_output=True,
+    )
+    deployer_balance = int(res.stdout.strip())
```
Added this balance check

refcell committed:
Fix contract indices nit

OptimismBot committed:
Merge pull request #7316 from ethereum-optimism/refcell/constant-time-parent-lookup

fix(op-challenger): Constant Time Parent Lookup

mergify[bot] committed:
Merge branch 'develop' into refcell/complete-games-queue

dependabot[bot] committed:
build(deps): bump envalid from 7.3.1 to 8.0.0

Bumps [envalid](https://github.com/af/envalid) from 7.3.1 to 8.0.0.
- [Release notes](https://github.com/af/envalid/releases)
- [Commits](https://github.com/af/envalid/compare/v7.3.1...v8.0.0)

---
updated-dependencies:
- dependency-name: envalid
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Spacesai1or committed:
Init MM metrics

OptimismBot committed:
Merge pull request #7261 from ethereum-optimism/refcell/complete-games-queue

fix(op-challenger): Don't schedule updates for resolved games

mergify[bot] committed:
Merge branch 'develop' into refcell/executor-thread-metrics

OptimismBot committed:
Merge pull request #7262 from ethereum-optimism/refcell/executor-thread-metrics

feat(op-challenger): Concurrency Metrics

mergify[bot] committed:
Merge branch 'develop' into revert-indexer-disable

OptimismBot committed:
Merge pull request #7265 from ethereum-optimism/revert-indexer-disable

chore(indexer): reenable indexer fully on ci

roninjin10 committed:
Apply suggestions from code review

Co-authored-by: Hamdi Allam <hamdi.allam97@gmail.com>

mergify[bot] committed:
Merge branch 'develop' into wyatt/ufm/init-mm-metrics

Commit:
lint

OptimismBot committed:
Merge pull request #7320 from ethereum-optimism/wyatt/ufm/init-mm-metrics

Init Metamask Pass/Fail metrics

mergify[bot] committed:
Merge branch 'develop' into refcell/claim-count-tracker

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.5.0 to 20.6.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.5.0 to 20.6.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7264 from ethereum-optimism/refcell/claim-count-tracker

feat(op-challenger): Claim Count Metrics

mergify[bot] committed:
Merge branch 'develop' into 09-15-feat_indexer_Enable_devnet_indexer_as_a_preset

OptimismBot committed:
Merge pull request #7271 from ethereum-optimism/09-15-feat_indexer_Enable_devnet_indexer_as_a_preset

feat(indexer): Enable devnet indexer as a preset

OptimismBot committed:
Merge pull request #7247 from ethereum-optimism/09-14-chore_indexer_remove_indexer_all_command

chore(indexer): remove indexer all command

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/types/node-20.6.3

OptimismBot committed:
Merge pull request #7318 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.6.3

build(deps-dev): bump @types/node from 20.5.0 to 20.6.3

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/envalid-8.0.0

dependabot[bot] committed:
build(deps-dev): bump vitest from 0.34.1 to 0.34.4

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 0.34.1 to 0.34.4.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v0.34.4/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump glob from 10.3.3 to 10.3.5

Bumps [glob](https://github.com/isaacs/node-glob) from 10.3.3 to 10.3.5.
- [Changelog](https://github.com/isaacs/node-glob/blob/main/changelog.md)
- [Commits](https://github.com/isaacs/node-glob/compare/v10.3.3...v10.3.5)

---
updated-dependencies:
- dependency-name: glob
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7326 from ethereum-optimism/dependabot/npm_and_yarn/vitest-0.34.4

build(deps-dev): bump vitest from 0.34.1 to 0.34.4

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/glob-10.3.5

OptimismBot committed:
Merge pull request #7327 from ethereum-optimism/dependabot/npm_and_yarn/glob-10.3.5

build(deps-dev): bump glob from 10.3.3 to 10.3.5

jbearer commented on op-node/rollup/derive/batches.go:
```diff
@@ -139,7 +148,14 @@ func CheckBatchEspresso(cfg *rollup.Config, log log.Logger, l2SafeHead eth.L2Blo
+		log.Warn("error reading next possible L1 origin", "err", err, "origin", nextL1Number)
+		return BatchUndecided
+	}
+	nextL1BlockEligible := nextL1Block.Time <= cfg.BlockTime+l2SafeHead.Time
```
It might be cleaner to encapsulate this logic in `EspressoL1Origin`, so it just takes the next `L1BlockRef`, not a `bool`. Then there's no chance of bugs from different callers having a different idea of what "eligible" means.

nomaxg replied:
agreed this feels better

jbearer commented on ops-bedrock/demo-docker-compose.yml:
```diff
@@ -0,0 +1,263 @@
+  op_blockscout_logs:
+  op_blockscout_redis:
+
+services:
```
At some point we should clean up all the various docker-compose files by having one definition of each service and using `extends` to bring each service into each compose setup. But this is not urgent.

jbearer commented on packages/contracts-bedrock/deploy-config/espresso-demo.json:
```diff
@@ -0,0 +1,54 @@
+{
```
For the other devnets we don't commit this generate file, only the template. I think we should `git rm` this and add it to `packages/contracts-bedrock/.gitignore`. Unless there's a specific reason to version control it

protolambda committed:
contracts-bedrock: internal devnet ProtocolVersions deployment

protolambda committed:
go: update libp2p to fix Go 1.21 build

tynes committed:
Merge pull request #7332 from ethereum-optimism/deploy-devnet-protocol-versions

contracts-bedrock: internal devnet ProtocolVersions deployment

tynes committed:
Merge pull request #7319 from ethereum-optimism/dependabot/npm_and_yarn/envalid-8.0.0

build(deps): bump envalid from 7.3.1 to 8.0.0

mergify[bot] committed:
Merge branch 'develop' into update-libp2p

OptimismBot committed:
Merge pull request #7333 from ethereum-optimism/update-libp2p

go: update libp2p to fix Go 1.21 build

nomaxg committed:
unformat init.py, pass l1blockref down to the espresso l1 origin function, misc gitignore cleanup

nomaxg committed:
Merge branch 'integration' into espresso-demo-setup

jbearer reviewed (commented):
Changes look good. Any idea what's going on with the CI failures?

nomaxg commented:
> Changes look good. Any idea what's going on with the CI failures?

Looks like some strange version issue. quic-go is incompatible with go 1.21 apparently?

nomaxg commented:
I think that this may be the culprit, I'll try pinning this to 1.20:

https://github.com/EspressoSystems/op-espresso-integration/commit/60fd40ab2a4728e23da738091886a6a52280ebbe#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R815

nomaxg commented:
> I think that this may be the culprit, I'll try pinning this to 1.20:
> 
> [60fd40a#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R815](https://github.com/EspressoSystems/op-espresso-integration/commit/60fd40ab2a4728e23da738091886a6a52280ebbe#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R815)

Hmm actually just calls go test under the hood, not sure what the issue is

nomaxg commented:
Here we go: https://github.com/ethereum-optimism/optimism/pull/7333

nomaxg committed:
update libp2p

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=76) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=76) before we can accept your contribution.<br/>**1** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: nomaxg<br/>:x: karlb<br/>:x: OptimismBot<br/>:x: mergify[bot]<br/>:x: spacesailor24<br/>:x: refcell<br/>:x: roninjin10<br/>:x: Will Cory<br/>:x: tynes<br/>:x: protolambda<br/><hr/>**Will Cory** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/op-espresso-integration?pullRequest=76) it.</sub>

nomaxg committed:
Revert "update libp2p"

This reverts commit 7188b329126b6ec942c741f86477add65a809b1a, reversing
changes made to 38449bbce863ad94f4b33cfd12de9fd79ce4b4d8.

jbearer reviewed (approved):
Ok, approving, please just make sure the build is fixed before merging

nomaxg committed:
update libp2p

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=76) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=76) before we can accept your contribution.<br/>**1** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: nomaxg<br/>:x: mergify[bot]<br/>:x: OptimismBot<br/>:x: karlb<br/>:x: roninjin10<br/>:x: refcell<br/>:x: spacesailor24<br/>:x: tynes<br/>:x: protolambda<br/>:x: Will Cory<br/><hr/>**Will Cory** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/op-espresso-integration?pullRequest=76) it.</sub>

nomaxg committed:
add back ff dep

nomaxg committed:
lint fixes
